app/questionnaire/dependency_builder.py

```diff
@@ -0,0 +1,74 @@
+from app.questionnaire.dependencies import Dependencies
+
+
+class DependenciesBuilder:
```

Comment:
Since the dependencies state is held and managed by the `Dependencies` type I don't think there's much value in this being a class itself.

In a static language this would be a static class - it holds no state. In python I'd suggest that it's more pythonic to define the methods as functions on the module instead.